cpp/support/container.h

```diff
@@ -0,0 +1,146 @@
+#ifndef XGRAMMAR_SUPPORT_CONTAINER_H_
+#define XGRAMMAR_SUPPORT_CONTAINER_H_
+#include <vector>
+
+#include "logging.h"
+
+namespace xgrammar {
+
+namespace details {
+
+template <typename Node>
+class NodePool {
+ public:
+  NodePool() = default;
+
+  void reserve(int n) { node_pool_.reserve(n); }
```

Comment:
```suggestion
  void Reserve(int n) { node_pool_.reserve(n); }
```
All methods should be captialized